prusa/connect/printer/__init__.py

```diff
@@ -682,6 +691,28 @@ def loop(self):
             except Exception:
                 log.exception("Unexpected exception caught in SDK loop!")
 
+    def snapshot_loop(self):
+        """Gets an item Snapshot from queue and sends it"""
+        while self.__running_loop:
+            try:
+                # Get the item to send
+                item = \
+                    self.snapshot_queue.get(timeout=const.TIMESTAMP_PRECISION)
+
+                # Send it
+                res = item.send_data(self.conn, self.server)
+                if res.status_code in (401, 403):
+                    item.fail_cb()
+                if res.status_code > 400:
+                    log.warning(res.text)
+                elif res.status_code == 400:
+                    log.debug(res.text)
+            except Empty:
+                return
```

Comment:
I think that there should be `continue` instead of `return` ....

Reply:
original loop has return here

Reply:
no original loop, but loop_step :-)